lib/setupmanagers/qemuhck/qemu_machine.rb

```diff
@@ -374,9 +374,25 @@ def full_replacement_map
                          @define_variables)
     end
 
+    def read_dynamic_device(device)
+      file_path = Pathname.new(DEVICES_JSON_DIR).join("#{device}.rb")
```

Comment:
Call `realpath` here instead of after `File.exist?` so that we don't have to resolve symlinks twice.

Reply:
we cant do that, _.realpath_ assume that file exists

Reply:
Then let's rescue `Errno::ENOENT` for `realpath` instead of having another `File.exist?` check.